syft/frameworks/torch/tensors/interpreters/additive_shared.py

```diff
@@ -462,6 +456,28 @@ def __getitem__(self, indices):
 
     ## SECTION SPDZ
 
+    def _convert_to_shared_tensor(self, other):
+        if isinstance(other, int):
+            other = torch.tensor([other], dtype=self.torch_dtype)
+
+        def share(other):
+            return other.share(
+                *self.child.keys(),
+                field=self.field,
+                dtype=self.dtype,
+                crypto_provider=self.crypto_provider,
+                **no_wrap,
```

Comment:
oh btw there is a new attribute which is missing (independently of your work), it's `protocol=...`!
So avoid this in the future you can use the `def get_class_attributes` method! Usage will be here: `.share(**self.get_class_attributes(), **no_wrap)`

Reply:
I changed as you said and deduplicated all patterns used in this file using the `_to_shared_tensor`. ^^7